src/user-guide/writing-workflows/runtime.rst

```diff
@@ -615,61 +709,130 @@ event handlers using the alternate methods:
                cylc message -- "${CYLC_WORKFLOW_ID}" "${CYLC_TASK_JOB}" 'oopsy daisy'
            """
            execution retry delays = PT0S, PT30S
-           [[[events]]]
-               handlers = "echo '!!!!!EVENT!!!!!' "
+           [[[events]]]  # general handlers:
+               handlers = notify-events.py
                # Note: task output name can be used as an event in this method
                handler events = retry, failed, oops
            [[[outputs]]]
                oops = oopsy daisy
 
-The handler command here - specified with no arguments - is called with the
-default arguments, like this:
+Built-in Email Event Handler
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To send an email on task events, configure relevant tasks with a list of events
+to handle by email. Custom task output names can also be used as event names,
+in which case the event triggers when the output message is received.
+
+E.g. to send an email on task failed, retry, and a custom message event:
+
+.. code-block:: cylc
+
+   [runtime]
+       [[foo]]
+           script = """
+               test ${CYLC_TASK_TRY_NUMBER} -eq 3
+               cylc message -- "${CYLC_WORKFLOW_ID}" "${CYLC_TASK_JOB}" 'oopsy daisy'
+           """
+           execution retry delays = PT0S, PT30S
+           [[[events]]]
+               mail events = failed, retry, oops
+           [[[outputs]]]
+               oops = oopsy daisy
+
+By default, event emails will be sent to the current user with:
+
+- ``to:`` set as ``$USER``
+- ``from:`` set as ``notifications@$(hostname)``
+- SMTP server at ``localhost:25``
 
-.. code-block:: bash
+These can be configured using the settings:
 
-   echo '!!!!!EVENT!!!!!' %(event)s %(workflow)s %(id)s %(message)s
+.. cylc-scope:: flow.cylc[runtime][<namespace>]
+
+- :cylc:conf:`[mail]to` (list of email addresses)
+- :cylc:conf:`[mail]from`
+
+.. cylc-scope::
+
+The scheduler batches events over a 5 minute interval, by default, to avoid
+flooding your Inbox if many events occur in a short time. The batching interval
+can be configured with :cylc:conf:`[scheduler][mail]task event batch interval`.
 
 
 .. _Late Events:
 
 Late Events
 ^^^^^^^^^^^
 
-You may want to be notified when certain tasks are running late in a real time
-production system - i.e. when they have not triggered by *the usual time*.
-Tasks of primary interest are not normally clock-triggered however, so their
-trigger times are mostly a function of how the workflow runs in its environment,
-and even external factors such as contention with other workflows [1]_ .
+.. warning::
+
+  The scheduler can only check for lateness once a task has appeared in its
+  active task window. In Cylc 8 this is usually when the task is actually
+  ready to run, which severely limits the usefulness of late events as
+  currently implemented.
```

Comment:
```suggestion
   The scheduler can only check for lateness once a task has appeared in its
   active task window. In Cylc 8 this is usually when the task is actually
   ready to run, which severely limits the usefulness of late events as
   currently implemented.
```

Reply:
Sphinx doesn't mind, right? Is 3 spaces a convention we've agreed on?

Reply:
(applied anyway)

Reply:
@oliver-sanders Is it worth adding that to CONTRIBUTING.md?

Reply:
Yeah, in a slightly more refined form.

I think this is in the Sphinx docs and in the rose/developing page (I think the rose dev pages are a little out of date).